arrow-cpp: 0.12.0 -> 0.13.0

veprbl opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

veprbl committed:
snappy: fix include path in generated cmake target

veprbl committed:
arrow-cpp: 0.12.0 -> 0.13.0
pythonPackages.pyarrow: 0.12.0 -> 0.13.0

*_HOME variables are now gone. There are similar *_ROOT variables, but
the package discovery currently works without that.

veprbl commented:
@GrahamcOfBorg build python2Packages.pyarrow python3Packages.pyarrow

veprbl committed:
uriparser: enable build on darwin

veprbl commented:
@GrahamcOfBorg build python2Packages.pyarrow python3Packages.pyarrow

pSub reviewed (approved):
LGTM

xeji commented:
aarch64 should build now
@GrahamcOfBorg build arrow-cpp python2Packages.pyarrow, python3Packages.pyarrow

veprbl commented:
@GrahamcOfBorg build python2Packages.pyarrow python3Packages.pyarrow

veprbl committed:
arrow-cpp: fix build on architectures other than x86_64

Settting ARROW_USE_SIMD=OFF used to fix i686 build for 1.12.0.
The fix for ARROW-5007 should help on aarch64.

veprbl commented:
@GrahamcOfBorg build python2Packages.pyarrow python3Packages.pyarrow
